mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -1332,20 +1332,31 @@ public void onJobSubmit(final SubmitJobRequest request) {
 
     private JobDefinition getResolvedJobDefinition(final String user, final Optional<JobDefinition> givenJobDefnOp) throws Exception {
         JobDefinition resolvedJobDefn;
-        if(givenJobDefnOp.isPresent()) {
-            resolvedJobDefn = givenJobDefnOp.get();
-        }
-        else {
+        if(!givenJobDefnOp.isPresent()) {
             // no job definition specified , this is quick submit which is supposed to inherit from last job submitted
-            List<JobInfo> existingJobsList = new ArrayList<>(jobManager.getAllNonTerminalJobsList());
+            List<JobInfo> existingJobsList = jobManager.getAllNonTerminalJobsList();
             Optional<JobDefinition> jobDefnOp = createNewJobDefinitionFromLastSubmittedInheritSchedInfoAndParameters(existingJobsList, jobManager.getCompletedJobsList(), empty(),jobStore);
             if(jobDefnOp.isPresent()) {
                 logger.info("Inherited scheduling Info and parameters from previous job");
                 resolvedJobDefn = jobDefnOp.get();
             } else {
                 throw new Exception("Job Definition could not retrieved from a previous submission (There may not be a previous submission)");
             }
-
+        }
+        else if (givenJobDefnOp.get().getSchedulingInfo() != null && givenJobDefnOp.get().getSchedulingInfo().requireInheritInstanceCheck()) {
```

Comment:
nit: Did you considering making requireInheritanceCheck a property of JobDefinition instead of SchedulingInfo?